code/search/src/binary_search/binary_search.py

```diff
@@ -3,26 +3,29 @@
 '''
 
 
-def binarySearchRecursive(arr, l, r, x):
-    if r >= l:
-
+def _binary_search_recursive_impl(arr, x, l, r):
+    if r > l:
         mid = l + int((r - l) / 2)
 
         if arr[mid] == x:
             return mid
 
         elif arr[mid] > x:
-            return binarySearchRecursive(arr, l, mid - 1, x)
+            return _binary_search_recursive_impl(arr, x, l, mid - 1)
```

Comment:
According to the [PEP 8](https://www.python.org/dev/peps/pep-0008/) style guide, I don't think you should be using a leading underscore for the name.

Reply:
I think the method is private and [PEP 8 - Method names "Use one leading underscore only for non-public methods"](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables), thus I added leading underscore.